batcher/src/sync.rs

```diff
@@ -21,11 +21,26 @@ This method spawns a background thread and runs [`Receiver::exec`] on it. The ha
 pub fn spawn<T: Channel + Send + 'static>(
     thread_name: impl Into<String>,
     receiver: Receiver<T>,
-    mut on_batch: impl FnMut(T) -> Result<(), BatchError<T>> + Send + 'static,
+    on_batch: impl FnMut(T) -> Result<(), BatchError<T>> + Send + 'static,
 ) -> io::Result<thread::JoinHandle<()>>
 where
     T::Item: Send + 'static,
 {
+    #![allow(unreachable_code)]
+
+    #[cfg(all(
+        target_arch = "wasm32",
+        target_vendor = "unknown",
+        target_os = "unknown"
+    ))]
+    {
+        let _ = (thread_name, receiver, on_batch);
+
+        panic!("blocking channel spawning is not supported on this platform");
```

Comment:
These methods are fallible, so they should just fail. We also need to document behavior on wasm